nf_core/lint.py

```diff
@@ -68,19 +68,41 @@ def run_linting(pipeline_dir, release_mode=False):
     if len(lint_obj.failed) > 0:
         logging.error(
             "Sorry, some tests failed - exiting with a non-zero error code...{}\n\n"
-            .format("\n       Reminder: Lint tests were run in --release mode." if release else '')
+            .format("\n\tReminder: Lint tests were run in --release mode." if release_mode else '')
         )
 
     return lint_obj
 
 
 class PipelineLint(object):
-    """ Object to hold linting info and results """
-
-    def __init__(self, pipeline_dir):
+    """Object to hold linting information and results.
+    All objects attributes are set, after the :meth:`PipelineLint.lint_pipeline` function was called.
+
+    Attributes:
+        conda_config (dict): The parsed conda configuration file content (``environment.yml``). The dictionary has
+                             the following structure, with some example values::
+
+                                 {'name': 'nf-core-hlatyping',
+                                  'channels': ['bioconda', 'conda-forge'],
+                                  'dependencies': ['optitype=1.3.2', 'yara=0.9.6']}
+
+        conda_package_info (dict):
+        config (dict):
+        dockerfile (list):
+        failed (list):
```

Comment:
of course if there is some convention based on Markdown, from which API docs can be generated, I am happy to give it a try.